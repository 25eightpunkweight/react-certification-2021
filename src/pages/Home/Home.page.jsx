import React, { useRef } from 'react';
import { Link, useHistory } from 'react-router-dom';

import { useAuth } from '../../providers/Auth';
import './Home.styles.css';
import mockData from '../../../public/youtube-videos-mock.json';
import CardItem from '../../components/CardItem'
import Header from '../../components/Header';

function HomePage() {
  const history = useHistory();
  const sectionRef = useRef(null);
  // const { authenticated, logout } = useAuth();
  const trimmedArray = mockData;//.items.slice(0, 5);

  function deAuthenticate(event) {
    event.preventDefault();
    logout();
    history.push('/');
  }

  return (
    <section className="homepage" ref={sectionRef}>
      {/* <h1>Hello stranger!</h1>
      {authenticated ? (
        <>
          <h2>Good to have you back</h2>
          <span>
            <Link to="/" onClick={deAuthenticate}>
              ← logout
            </Link>
            <span className="separator" />
            <Link to="/secret">show me something cool →</Link>
          </span>
        </>
      ) : (
        <Link to="/login">let me in →</Link>
      )} */}
      <div className="cards">{trimmedArray.map((d) => CardItem(d))}</div>
    </section>

  );
}

export default HomePage;
