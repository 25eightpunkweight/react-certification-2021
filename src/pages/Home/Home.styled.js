import styled from 'styled-components';

const HomePageWrapper = styled.section`
  text-align: center;
  overflow: auto;

  &::-webkit-scrollbar {
    display: none;
  }

  & > h1 {
    font-size: 3rem;
    letter-spacing: -2px;
  }
`;

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-top: 1rem;
  margin: 3px;
  padding: 20px;
  overflow: auto;
`;

const Cards = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-top: 3px;
  margin-left: auto;
  margin-right: auto;
  padding: 20px;
  overflow: auto;
  align-items: center;
`;

const Styled = {
  HomePageWrapper,
  Container,
  Cards,
};

export default Styled;