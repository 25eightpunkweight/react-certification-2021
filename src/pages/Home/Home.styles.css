.homepage {
  text-align: center;
  overflow: auto;
}

.homepage h1 {
  font-size: 3rem;
  letter-spacing: -2px;
}

div.cards {
  display: flex;
  flex-wrap: wrap;
  margin: 3px;
  padding: 20px;
  overflow: scroll;
}