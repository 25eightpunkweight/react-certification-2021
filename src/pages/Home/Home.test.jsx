import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home.page';
import TestAppContainer from '../../test/TestAppContainer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <TestAppContainer>
        <Home />
      </TestAppContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
