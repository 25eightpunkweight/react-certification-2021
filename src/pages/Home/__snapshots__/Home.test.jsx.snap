// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  className="sc-gKAaRy lNXoH"
  src="https://media.tenor.com/images/a742721ea2075bc3956a2ff62c9bfeef/tenor.gif"
/>
`;
